lib/Resizable.js

```diff
@@ -201,27 +186,24 @@ export default class Resizable extends React.Component<Props, State> {
       }
 
       // Update w/h
-      let width = this.state.width + (canDragX ? deltaX : 0);
-      let height = this.state.height + (canDragY ? deltaY : 0);
+      let width = this.props.width + (canDragX ? deltaX : 0);
+      let height = this.props.height + (canDragY ? deltaY : 0);
 
       // Early return if no change
-      const widthChanged = width !== this.state.width, heightChanged = height !== this.state.height;
+      const widthChanged = width !== this.props.width, heightChanged = height !== this.props.height;
       if (handlerName === 'onResize' && !widthChanged && !heightChanged) return;
 
       [width, height] = this.runConstraints(width, height);
 
       // Set the appropriate state for this handler.
       const newState = {};
       if (handlerName === 'onResizeStart') {
-        newState.resizing = true;
+        // nothing
```

Comment:
looks like we can rewrite `if` statement:

```typescript
if (handlerName === 'onResizeStop') {
  newState.slackW = newState.slackH = 0;
} else if (handlerName !== 'onResizeStart') {
  if (width === this.props.width && height === this.props.height) return;
}
```

Reply:
Yeah - this is intentional just to enumerate all the possible handlers for readability in the future.